test/lib/templateFactory.test.js

```diff
@@ -445,6 +459,234 @@ describe('Template Factory', () => {
+                {
+                    id: 8,
+                    templateId: 10
+                },
+                {
+                    id: 9,
+                    templateId: 1
+                },
+                {
+                    id: 10,
+                    templateId: 4
+                }
+            ];
+
+            returnValue = [
+                {
+                    id: 1,
+                    name,
+                    namespace,
+                    version: '1.0.1',
+                    metrics: {
+                        jobs: {
+                            count: 3
+                        },
+                        builds: {
+                            count: 4
+                        }
+                    }
+                },
+                {
+                    id: 3,
+                    name,
+                    namespace,
+                    version: '1.0.3',
+                    metrics: {
+                        jobs: {
+                            count: 0
+                        },
+                        builds: {
+                            count: 0
+                        }
+                    }
+                },
+                {
+                    id: 2,
+                    name,
+                    namespace,
+                    version: '1.0.2',
+                    metrics: {
+                        jobs: {
+                            count: 0
+                        },
+                        builds: {
+                            count: 0
+                        }
+                    }
+                },
+                {
+                    id: 4,
+                    name: `${namespace}/${name}`,
+                    version: '1.0.2',
+                    metrics: {
+                        jobs: {
+                            count: 2
+                        },
+                        builds: {
+                            count: 3
+                        }
+                    }
+                }
+            ];
+        });
+
+        /* it('should list templates with metrics when namespace is passed in', () => {
+            expected = [returnValue[0], returnValue[1], returnValue[2]];
+            datastore.scan.resolves(expected);
+            buildFactoryMock.list.resolves(buildsMock);
+            jobFactoryMock.list.resolves(jobsMock);
+
+            return factory.listWithMetrics(config).then((templates) => {
+                let i = 0;
+
+                templates.forEach((t) => {
+                    assert.deepEqual(t.id, expected[i].id);
+                    assert.deepEqual(t.metrics.jobs.count, expected[i].metrics.jobs.count);
+                    assert.deepEqual(t.metrics.builds.count, expected[i].metrics.builds.count);
+                    i += 1;
+                });
+            });
+        }); */
+
+        it('should list templates with metrics when no namespace is passed in', () => {
+            expected = [returnValue[3]];
```

Comment:
it's better to add this line assert.expect(x); with x being number of asserts in your test

Reply:
assert.expect() is not a function in this repo